galite-core/src/main/kotlin/org/kopi/galite/visual/VMenuTree.kt

```diff
@@ -18,8 +18,432 @@
+    private const val SELECT_GROUP_RIGHTS_BY_USERID = "" // TODO
+    private const val SELECT_GROUP_RIGHTS_BY_GROUPID = "" // TODO
+    const val CMD_QUIT = 0
+    const val CMD_OPEN = 1
+    const val CMD_SHOW = 2
+    const val CMD_ADD = 3
+    const val CMD_REMOVE = 4
+    const val CMD_FOLD = 5
+    const val CMD_UNFOLD = 6
+    const val CMD_INFORMATION = 7
+    const val CMD_HELP = 8
+    const val MAIN_MENU = -1
+    const val USER_MENU = -2
+    const val ADMIN_MENU = -3
+    const val BOOKMARK_MENU = -4
+    private val ROOT_MENUS = arrayOf(
+            RootMenu(MAIN_MENU, "forms"),
+            RootMenu(USER_MENU, "user"),
+            RootMenu(ADMIN_MENU, "admin"))
+    private const val MENU_LOCALIZATION_RESOURCE = "resource/org/kopi/galite/Menu"
+
+    init {
+      WindowController.windowController.registerWindowBuilder(Constants.MDL_MENU_TREE, object : WindowBuilder {
+        override fun createWindow(model: VWindow): UWindow {
+          return UIFactory.uiFactory.createView(model) as UMenuTree
+        }
+      })
+    }
+  }
+
+  // --------------------------------------------------------------------
+  // DATA MEMBERS
+  // --------------------------------------------------------------------
+  var root: TreeNode? = null
+    private set
+  private val actors = arrayOfNulls<VActor>(9)
+  lateinit var moduleArray: Array<Module> // Sets the accessibility of the module
+    private set
+  private val items = mutableListOf<Module>()
+  private val shortcutsID = mutableListOf<Int>()
+
+  // ----------------------------------------------------------------------
+  // CONSTRUCTORS
+  // ----------------------------------------------------------------------
+  init {
+    createActor(CMD_QUIT, "File", "Close", "quit", 0 /*KeyEvent.VK_ESCAPE*/, 0)
+    createActor(CMD_OPEN, "Edit", "Open", "open", KeyEvent.VK_ENTER, 0)
+    createActor(CMD_SHOW, "Edit", "Show", null, 0, 0)
+    createActor(CMD_ADD, "Edit", "Add", null, 0, 0)
+    createActor(CMD_REMOVE, "Edit", "Remove", null, 0, 0)
+    createActor(CMD_FOLD, "Edit", "Fold", "fold", KeyEvent.VK_ENTER, 0)
+    createActor(CMD_UNFOLD, "Edit", "Unfold", "unfold", KeyEvent.VK_ENTER, 0)
+    createActor(CMD_INFORMATION, "Help", "Information", null, 0, 0)
+    createActor(CMD_HELP, "Help", "Help", "help", KeyEvent.VK_F1, 0)
+    setActors(actors.filterIsInstance<VActor>().toTypedArray())
+    localizeActors(ApplicationContext.getDefaultLocale())
+    createTree(isSuperUser || loadFavorites)
+    localizeRootMenus(ApplicationContext.getDefaultLocale())
+  }
+
+  /**
+   * Constructs a new instance of VMenuTree.
+   * @param ctxt the context where to look for application
+   */
+  constructor(ctxt: DBContext) : this(ctxt, false, null, true) {}
+
+
+  constructor(ctxt: DBContext,
+              isSuperUser: Boolean,
+              userName: String?,
+              loadFavorites: Boolean)
+          : this(ctxt, isSuperUser, userName, null, loadFavorites)
+
```

Comment:
we can remove them and use the principal constructor to optimize

Reply:
yes, I did